include/gz/physics/ConeShape.hh

```diff
@@ -0,0 +1,151 @@
+      };
+
+      public: template <typename PolicyT>
+      class Implementation : public virtual Feature::Implementation<PolicyT>
+      {
+        public: using Scalar = typename PolicyT::Scalar;
+
+        public: virtual Scalar GetConeShapeRadius(
+            const Identity &_coneID) const = 0;
+
+        public: virtual Scalar GetConeShapeHeight(
+            const Identity &_coneID) const = 0;
+      };
+    };
+
+    /////////////////////////////////////////////////
+    /// \brief This feature sets the ConeShape properties such as
+    /// the cone radius and height.
+    class GZ_PHYSICS_VISIBLE SetConeShapeProperties
+        : public virtual FeatureWithRequirements<ConeShapeCast>
+    {
+      public: template <typename PolicyT, typename FeaturesT>
+      class ConeShape : public virtual Entity<PolicyT, FeaturesT>
+      {
+        public: using Scalar = typename PolicyT::Scalar;
+
+        /// \brief Set the radius of this ConeShape
+        /// \param[in] _radius
+        ///   The desired radius of this ConeShape
+        public: void SetRadius(Scalar _radius);
+
+        /// \brief Set the height of this ConeShape
+        /// \param[in] _height
+        ///   The desired height of this ConeShape
+        public: void SetHeight(Scalar _height);
+      };
+
+      public: template <typename PolicyT>
+      class Implementation : public virtual Feature::Implementation<PolicyT>
+      {
+        public: using Scalar = typename PolicyT::Scalar;
+
+        public: virtual void SetConeShapeRadius(
+            const Identity &_coneID, Scalar _radius) = 0;
+
+        public: virtual void SetConeShapeHeight(
+            const Identity &_coneID, Scalar _height) = 0;
+      };
+    };
+
+    /////////////////////////////////////////////////
+    /// \brief This feature constructs a new cone shape and attaches the
+    /// desired pose in the link frame. The pose could be defined to be the
+    /// cone center point in actual implementation.
+    class GZ_PHYSICS_VISIBLE AttachConeShapeFeature
+        : public virtual FeatureWithRequirements<ConeShapeCast>
+    {
+      public: template <typename PolicyT, typename FeaturesT>
+      class Link : public virtual Feature::Link<PolicyT, FeaturesT>
+      {
+        public: using Scalar = typename PolicyT::Scalar;
+
+        public: using PoseType =
+            typename FromPolicy<PolicyT>::template Use<Pose>;
+
+        public: using ShapePtrType = ConeShapePtr<PolicyT, FeaturesT>;
+
+        /// \brief Rigidly attach a ConeShape to this link.
+        /// \param[in] _radius
+        ///   The radius of the cone.
+        /// \param[in] _height
+        ///   The height of the cone.
+        /// \param[in] _pose
+        ///   The desired pose of the ConeShape relative to the Link frame.
+        /// \returns a ShapePtrType to the newly constructed ConeShape
+        public: ShapePtrType AttachConeShape(
+            const std::string &_name = "cone",
```

Comment:
include `<string>`

Reply:
I think that is already included:
https://github.com/gazebosim/gz-physics/blob/fa0b698b2faf3c08fe2bace4e7e3b951af982b65/include/gz/physics/ConeShape.hh#L21